tests/integration/test_integration.py

```diff
@@ -36,6 +43,60 @@ async def deploy_charm(ops_test: OpsTest, charm: Path) -> None:
             num_units=5,
         )
 
+    async def _get_vault_endpoint(self, ops_test: OpsTest) -> str:
+        """Retrieves the Vault endpoint by using Traefik's `show-proxied-endpoints` action.
+
+        Args:
+            ops_test: Ops test Framework.
+
+        Returns:
+            vault_endpoint: Vault proxied endpoint by Traefik.
+
+        Raises:
+            TimeoutError: If proxied endpoints are not retrieved.
+        """
+        traefik = ops_test.model.applications[TRAEFIK_APP_NAME]  # type: ignore[union-attr]
+        traefik_unit = traefik.units[0]
+        t0 = time.time()
+        timeout = 300  # seconds
```

Comment:
Why so long? Once traefik is in active/idle and that the relation is created it shouldn't take 5 min afaik

Reply:
I have put it as a precaution, we can decrease the timeout to 60 seconds.

Reply:
Done